Source/RunActivity/Viewer3D/Cameras.cs

```diff
@@ -1885,48 +1899,202 @@ public override void Update(ElapsedTime elapsedTime)
             targetLocation = attachedCar.WorldPosition.WorldLocation;
 
             // Train is close enough if the last car we used is part of the same train and still close enough.
-            var trainClose = (LastCheckCar != null) && (LastCheckCar.Train == train) && (WorldLocation.GetDistance2D(LastCheckCar.WorldPosition.WorldLocation, cameraLocation).Length() < MaximumDistance);
+            var trainClose = (LastCheckCar != null) && (LastCheckCar.Train == train) && (WorldLocation.GetDistance2D(LastCheckCar.WorldPosition.WorldLocation, cameraLocation).Length() < (SpecialPointFound ? MaximumSpecialPointDistance * 0.8f : MaximumDistance));
 
             // Otherwise, let's check out every car and remember which is the first one close enough for next time.
             if (!trainClose)
             {
                 foreach (var car in train.Cars)
                 {
-                    if (WorldLocation.GetDistance2D(car.WorldPosition.WorldLocation, cameraLocation).Length() < MaximumDistance)
+                    if (LastCheckCar != null && car == LastCheckCar && 
+                        WorldLocation.GetDistance2D(car.WorldPosition.WorldLocation, cameraLocation).Length() < (SpecialPointFound ? MaximumSpecialPointDistance * 0.8f : MaximumDistance))
                     {
+                        trainClose = true;
+                        break;
+                    }
+                    else if (WorldLocation.GetDistance2D(car.WorldPosition.WorldLocation, cameraLocation).Length() < MaximumDistance)
+                        {
                         LastCheckCar = car;
                         trainClose = true;
                         break;
                     }
                 }
+                if (!trainClose)
+                    LastCheckCar = null;
+            }
+            var trySpecial = false;
+            DistanceRunM += elapsedTime.ClockSeconds * train.SpeedMpS;
+            if (Math.Abs(DistanceRunM) >= CheckIntervalM)
+            {
+                DistanceRunM = 0;
+                if (!SpecialPointFound && SpecialPoints && trainClose) trySpecial = true;
             }
-
             // Switch to new position.
-            if (!trainClose || (TrackCameraLocation == WorldLocation.None))
-            {
-                var tdb = trainForwards ? new Traveller(train.FrontTDBTraveller) : new Traveller(train.RearTDBTraveller, Traveller.TravellerDirection.Backward);
-                tdb.Move(MaximumDistance * 0.75f);
-                var newLocation = tdb.WorldLocation;
-                TrackCameraLocation = new WorldLocation(newLocation);
-                var directionForward = WorldLocation.GetDistance((trainForwards ? train.FirstCar : train.LastCar).WorldPosition.WorldLocation, newLocation);
-                if (Viewer.Random.Next(2) == 0)
+            if (!trainClose || (TrackCameraLocation == WorldLocation.None) || trySpecial)
+            {
```

Comment:
The code in this block that locates the special locations is quite long and it isn't immediately clear why it is so complicated. Could you add some comments explaining what each small chunk is doing, e.g. what is `shortTrav` doing/for?

Reply:
Done. shortTrav is used to determine the direction of movement (left/right) of the camera from center of track to platform. I assume there is also another way to do this, but I couldn't find it.